File: js/arrays/0334-increasing-triplet-subsequence.js
var increasingTriplet = function(nums) {
    var m = Infinity;
    var n = Infinity;
    for (const num of nums) {
        if (num <= m) m = num;
        else if (num <= n) n = num;
        else return true;
    }
    
    return false;
};

File: rust/arrays/0334-increasing-triplet-subsequence.rs
fn increasing_triplet(nums: Vec<i32>) -> bool {
    let (mut i, mut j) = (i32::MAX, i32::MAX);
    for num in nums {
        if num <= i {
            i = num;
        } else if num <= j {
            j = num;
        } else {
            return true;
        }
    }

    false
}